### ISSUE TITLE
Version 1.0 does not work with Henkaku Enso

### ISSUE THREAD

**Issue by BoomerNT** [motoharu-gosuto/psvgamesd#31]
This issue can be reproduced with the following steps.

1. Start fresh from a 3.60 Vita without anything installed.
2. Fresh install of Henkaku 10.2 via browser.
3. Install psvgamesd 1.0. Confirm psvgamesd working.
4. Install Enso. System hang on Enso logo after reboot.
5. Hold L during boot to bypass plugins. Enso boots normally.

If Enso has been installed and working, adding psvgamesd in kernel plugins also prevents system to boot.

**Comment by Hking0036** [comment]
Not having any problems booting on enso, rebooted after plugin install. Are you sure your config is set up right?

**Comment by BoomerNT** [comment]
@Hking0036 

Yes, I used a freshly formatted Vita and freshly formatted memory card. Henkaku and Enso installed directly from official website.

After installing skprx and reboot, system will hang on Enso logo and auto shut down after a few seconds. Holding L to bypass skprx then Enso works.

Install skprx then launch Henkaku from website, everything works.

The only suspicious point is that I used a Japanese console (PCH-2000) and system language is not English.

**Comment by Hking0036** [comment]
I'm referring to the formatting in the config.txt, are you sure it's under *KERNEL and not another category?

**Comment by BoomerNT** [comment]
@Hking0036 

Yes, it is correct. Otherwise it won't work even if Henkaku is launched via browser.

Currently the issue on my side is that psvgamesd works properly when Henkaku is launched via browser, but it prevents Enso to boot.

**Comment by neededanametoreportbug** [comment]
Had similar issue.
I suspect a conflict between psvgamesd.skprx and nonpdrm.skprx, disabling one or the other allows enso to boot, but having both enabled makes the system freeze on the molecule bootlogo and then shut down.

**Comment by catchra** [comment]
I have nonpdrm and psvgamesd on my 3g model with enso and i have no problems.

**Comment by BoomerNT** [comment]
@neededanametoreportbug 

I set up everything fresh and nonpdrm does not even exist on my console. The only plugin besides Henkaku itself is psvgamesd. Then Enso hang up on logo and shut down.

**Comment by D-Pyro** [comment]
I had it hang after commenting out some plugins. If I remove the comments and just enable the plugins it won't hang.

**Comment by BoomerNT** [comment]
@D-Pyro 

That's an interesting find. In my case, without using any other plugins, Enso hangs on boot but browser entry point works OK. The difference is that Enso loads way earlier than browser. Perhaps some dependencies of psvgamesd is not loaded during Enso boot process but will be loaded after a clean boot and launching browser.

**Comment by vpauly94** [comment]
I can confirm the findings of @BoomerNT , I could only lauch it after unintalling Enso and using the browser exploit, changing the boot config didn't help on Enso either.

**Comment by catchra** [comment]
have you tryed useing ur0:tai instead of ux0:tai
i am useing ur0:tai for nonpdrm and psvitagamed on a 3G model without any problems

**Comment by neededanametoreportbug** [comment]
Perhaps this is a 2K model specific issue

**Comment by SPMelly** [comment]
Yea i have both nonpdrm and psvgamesd both enabled and im not having any trouble. Only tested on a phat though. Havent tried a slim yet

**Comment by BoomerNT** [comment]
@catchra 
I've tried with both ux0: and ur0: but neither worked.

@neededanametoreportbug 
Together with the comment of @SPMelly I think the problem might be narrowed down to compatibility between PS Vita slim (PCH-2000 series), Enso, and psvgamesd.

**Comment by D-Pyro** [comment]
My vita is a slim. Leaving all plugins enabled works for me. This includes sd2vita.

**Comment by Kalanyr** [comment]
I have what seems to be a variant of this problem. The kernel plugin causes sporadic freeze and  hard shutdown when on the molecule screen during boot.  The rates around ~50%. When it does boot everything seems to be fine.

(Slim, Australian non-3G Vita, running the plugin from ur0. Adrenaline and Browser Download plugins are also installed)

**Comment by motoharu-gosuto** [comment]
Thanks for feedback. Will see what can be done. Enso is not convinient for developement so I did not really test it with enso.

**Comment by hsuanpai** [comment]
If you have installed the ENSO, please Press + Hold L button and boot PSVita.
After boot, check function is working, then do ENSO install again and reboot.
Next boot, should be working normally.

I removed the SD2Vita adapter when I select virtual MMC and select ISO as mounted.
I will sometimes successfully launch the game when virtual MMC w/ ISO is mounted.

**Comment by vyper122** [comment]
Finally solved the issue(thank goodness!). Try to put 2-3 plugins under *KERNEL before psvgamesd on configtxt. I'm using PCH-2006(neon orange). please see example below:
*KERNEL
ux0:app/PSPEMUCFW/sce_module/adrenaline_kernel.skprx
ur0:tai/gamesd.skprx
ux0:tai/ds4vita.skprx
ux0:tai/psvgamesd.skprx
